src/ethereum/EthereumNetwork.ts

```diff
@@ -223,7 +223,7 @@ export class EthereumNetwork {
     return out
   }
 
-  getBaseFeePerGas = async (): Promise<string> => {
+  getBaseFeePerGas = async (): Promise<string | undefined> => {
```

Comment:
> Fix return type on getBaseFeePerGas

But what was wrong with the return type?

Reply:
I thought I explained this in a commit message but I don't see it. Plugins that don't support this rpc method return undefined because asyncWaterfall always returns undefined when passed an empty array.

Reply:
The undefined return value is actually expacted and handled correctly in a few places

Reply:
A more thorough fix to this issue would be to change ayncWaterfall's signature to:

``ts
export async function asyncWaterfall<T>(
  asyncFuncs: AsyncFunction[],
  timeoutMs: number = 5000
): Promise<T | undefined> {
```

And then fix all the places where there are type errors